### PR TITLE
Add video support

### PR DESCRIPTION
**Pull request by armpico**
This PR adds support to video stickers

### CODE REVIEW

File: picobot/video_editor.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    check_duration(duration)
+
+    (width, height) = estimate_video_sticker_size(width, height)
+    vid_output_path = mp4_path.with_suffix('.webm')
+    (
+        ffmpeg.input(mp4_path)
+        .filter('scale', width, height)
+        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
+        .run()
```
Comment (JPTIZ) [inline picobot/video_editor.py:26]: [2] Trying to `/addsticker` a video a second time as the same user makes `ffpmeg` ask to overwrite the file:

```console
File '/home/jptiz/dev/car/pico-bot/picobot/images/vid<my user ID>.webm' already exists. Overwrite? [y/N] y
```

Maybe one of these may work, according to [ffmpeg-python docs](https://kkroening.github.io/ffmpeg-python/#ffmpeg.overwrite_output):

1. Adding `.overwrite_output()` to ffmpeg builder:
```suggestion
        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
        .overwrite_output()
        .run()
```


2. ***Or*** adding `overwrite_output` kwarg to `.run`:
```suggestion
        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
        .run(overwrite_output=True)
```

File: picobot/video_editor.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    )
+    return vid_output_path
+
+
+def check_duration(duration: float):
+    if duration > 3.0:
+        raise Exception('video duration exceeds limits')
+
+
+def estimate_video_sticker_size(width: int, height: int):
```
Comment (JPTIZ) [inline picobot/video_editor.py:36]: [3] Annotate return types as well:

```suggestion
def estimate_video_sticker_size(width: int, height: int) -> tuple[int, int]:
```

File: picobot/video_editor.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        .filter('scale', width, height)
+        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
+        .run()
+    )
+    return vid_output_path
+
+
+def check_duration(duration: float):
+    if duration > 3.0:
+        raise Exception('video duration exceeds limits')
```
Comment (JPTIZ) [inline picobot/video_editor.py:33]: [4] A specific exception type would be more clear and easier to handle (e.g. for `except ValueError`):

```suggestion
        raise ValueError('Video duration exceeds limits')
```

(+ sneaky first letter upper 👀)

File: picobot/video_editor.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    '''
+    probe = ffmpeg.probe(mp4_path)
+    video_stream = next(
+        (stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None
+    )
+    width = int(video_stream['width'])
+    height = int(video_stream['height'])
+    duration = float(probe['format']['duration'])
+
+    check_duration(duration)
```
Comment (JPTIZ) [inline picobot/video_editor.py:18]: Hmmm I'm not sure I would split this into a function as it is literally just a simple straight-forward `if`. Personally I would bring the function's code here:

```suggestion
    if duration > 3.0:
            raise ValueError('Video duration exceeds limits')
```

**Review by JPTIZ — CHANGES_REQUESTED** [review]
:P I know you're working on some of these comments, but it is good to register them.

**Comment by JPTIZ** [comment]
Looks like we need a rebase :P

File: pyproject.toml [hunk 5]
```diff
@@ -6,11 +6,13 @@ authors = ["Arthur Mesquita Pickcius <arthur.pickcius@gmail.com>"]
 
 [tool.poetry.dependencies]
 python = "^3.7"
-python-telegram-bot = "^11.1"
+python-telegram-bot = "^13.11"
 dataset = "^1.1"
 pillow = "^8.0"
 python-slugify = "^4.0"
 emoji = "^0.6.0"
+typed-ast = "^1.5.2"
```
Comment (tarcisioe) [inline pyproject.toml:14]: This seems to be a transitive dependency and most likely shouldn't be here.

File: picobot/video_editor.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+import ffmpeg
+
+
+class VideoTooLongError(Exception):
```
Comment (tarcisioe) [inline picobot/video_editor.py:5]: Add a docstring.

File: picobot/handlers.py [hunk 7]
```diff
@@ -29,6 +33,10 @@
 DEFAULT_EMOJI = '😁'
 
 
+class FileTooLargeError(Exception):
```
Comment (tarcisioe) [inline picobot/handlers.py:36]: Add docstring.

File: picobot/video_editor.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+def sticker_from_video(video_path: Path, circle_mask: Path = None) -> Path:
+    '''
+    Converts the given video to a proper format that can be uploaded as a sticker.
+    Telegram accepts WEBM VP9 with a maximum size of 512x512 pixels with maximum 3 seconds duration.
+    '''
+    probe = ffmpeg.probe(video_path)
+    video_stream = next(
+        (stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None
+    )
+    width = int(video_stream['width'])
```
Comment (tarcisioe) [inline picobot/video_editor.py:18]: `video_stream` can be None. Check and abort early if this is the case.

File: picobot/video_editor.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    height = int(video_stream['height'])
+    duration = float(probe['format']['duration'])
+
+    if duration > 3.0:
+        raise VideoTooLongError('Video duration exceeds limits')
+
+    width, height = estimate_video_sticker_size(width, height)
+    vid_output_path = video_path.with_suffix('.webm')
+    vid_filename = video_path.name
```
Comment (tarcisioe) [inline picobot/video_editor.py:27]: Unused variable.

File: picobot/video_editor.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    in_file = ffmpeg.input(video_path).filter('scale', width, height)
+    if circle_mask is None:
+        (
```
Comment (tarcisioe) [inline picobot/video_editor.py:30]: Unnecessary parenthesis.

File: picobot/video_editor.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+            in_file.output(
+                filename=vid_output_path.as_posix(),
+                format='webm',
+                vcodec='libvpx-vp9',
+                pix_fmt='yuva420p',
+            ).run(overwrite_output=True)
+        )
+    else:
+        mask = ffmpeg.input(circle_mask).filter('alphaextract')
+        (
```
Comment (tarcisioe) [inline picobot/video_editor.py:40]: Unnecessary parethesis.

File: picobot/video_editor.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                vcodec='libvpx-vp9',
+                pix_fmt='yuva420p',
+            )
+            .run(overwrite_output=True)
+        )
+
+    return vid_output_path
+
+
+def estimate_video_sticker_size(width: int, height: int) -> tuple[int, int]:
```
Comment (tarcisioe) [inline picobot/video_editor.py:54]:
```suggestion
class VideoSize(NamedTuple):
    width: int
    height: int


def estimate_video_sticker_size(candidate_size: VideoSize) -> VideoSize:
```

**Review by tarcisioe — CHANGES_REQUESTED** [review]

**Review by JPTIZ — APPROVED** [review]
Moved some comments to issues:
- Files not deleted: https://github.com/caravelahc/pico-bot/issues/39.
- Explicit types in exception handling: https://github.com/caravelahc/pico-bot/issues/40.
- Lint/typing issues: https://github.com/caravelahc/pico-bot/issues/41.

**Pull request merged**